### ISSUE TITLE
CameraRoll.getAlbums() broken in Android API Level 29

### ISSUE THREAD

**Issue by dokkis** [react-native-community/react-native-cameraroll#174]
## Summary

CameraRoll.getAlbums() has 2 issues with API Level 29:
1. Exception thrown: `java.lang.IllegalArgumentException: Invalid column COUNT(*) as count.`
2. Exception get swallowed and doesn't return the promise here https://github.com/react-native-community/react-native-cameraroll/blob/master/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java#L430 so the function called CameraRoll.getAlbums() never get into the `then` or the `catch` part from client perspective

For the issue #1, it looks like GROUP BY support is removed from Android Q, that's why is not working.

see here:
* https://stackoverflow.com/questions/60623594/replacement-for-group-by-in-contentresolver-query-in-android-q-android-10-a
* https://stackoverflow.com/questions/468211/how-do-i-get-the-count-in-my-content-provider 

## Environment info

Android with API Level 29
Library version: 1.6.2

## Steps to reproduce

Try to use CameraRoll.getAlbums() on Android with API Level 29. No callback is never been called.

## Reproducible sample code

```
CameraRoll.getAlbums()
.then( (res) => console.log(res))
.catch( (err) => console.error(err))
```

No console statement is going to happen as there is an exception and has being swallowed without calling the reject callback.

**Comment by dokkis** [comment]
neither GROUP BY or DISTINCT would work on Android Q, the only way that I can think of to achieve this behaviour is having a cursor to scroll all `MediaStore.Images.ImageColumns.BUCKET_DISPLAY_NAME` and found the set of this values, then for each of them create a cursor with filter that will return the size using `getCount`.

I don't know how would look like performance-wise.

**Comment by r0b0t3d** [comment]
Just did some research. It seems the current solution doesn't works if you targeting api level 29 and above.
Found 2 keys in [ContentResolver](https://developer.android.com/reference/android/content/ContentResolver#QUERY_ARG_SQL_GROUP_BY) that are looks promising but they are added in Android R
```
QUERY_ARG_GROUP_COLUMNS
Specifies the list of columns (stored as a String[]) against which to group results.

QUERY_ARG_SQL_GROUP_BY
Key for an SQL style GROUP BY string that may be present in the query Bundle argument passed to ContentProvider#query(Uri, String[], Bundle, CancellationSignal).
```
So, the solution could be your suggestion above. But I think you just need 1 query for all media items in device, then iterate and group it by `BUCKET_DISPLAY_NAME`. Finally, count all items in each group.
You can also get the first item for thumbnail image for each album.
For now I don't have much time to work on it. Will check it when having free time. In the mean times, PR are welcome :D

**Comment by dokkis** [comment]
@r0b0t3d yep I am working to get together this PR hopefully today

**Comment by dokkis** [comment]
Created a PR here: https://github.com/react-native-community/react-native-cameraroll/pull/176

**Comment by bartolkaruza** [comment]
fixed by #176 

(unrelated, @dokkis you can let Github and semantic-release know that a PR fixes a particular issue by putting 
```
fixes #174
```
on a line in the description of the first commit. Then this issue would be auto closed and a comment that it is included in release x is added)

**Comment by dokkis** [comment]
@bartolkaruza thanks for the tip! I didn't knew that :)
